pirania/files/usr/libexec/rpcd/pirania

```diff
@@ -64,6 +70,56 @@ local function add_many_vouchers(msg)
     printJson(result)
 end
 
+local function add_visitor_voucher(msg)
+    local result = {}
+    local hostnameOutput = shell("cat /proc/sys/kernel/hostname")
+    local hostname = string.gsub(hostnameOutput, "%s+", "")
+    local noteHiph = string.gsub(msg.note, "%s+", "-")
+    local note = string.lower(noteHiph)
+    local secret = msg.secret
+    local epoc = msg.epoc
+    local key = hostname..'-visitor-'..note
+    result.hostname = hostname
+    result.key = key
+    local vouchers = ft.map(
+        function (val)
+            local randomKey = val.key
+            local newKey = key..'-'..randomKey
+            return json.encode(val)
+        end
+    , msg.vouchers)
+    local voucher = io.popen("voucher add_many_vouchers", 'w')
+    result.table = table.concat(vouchers, '\n')
+    -- result.success = assert(voucher:write(table.concat(vouchers, '\n')))
+    printJson(result)
+end
+
+local function add_member_voucher(msg)
+    local hostnameOutput = shell("cat /proc/sys/kernel/hostname")
```

Comment:
looks unnecesary long... could be splitted in smaller functions that contain part of the functionalty? looks it has many responsabilities